cmd/juju/model/destroy.go

```diff
@@ -216,6 +225,26 @@ func (c *destroyCommand) Run(ctx *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
+	// Check if the model has an sla auth.
+	if !slaIsSet {
+		return nil
+	}
+
+	bakeryClient, err := c.BakeryClient()
+	if err != nil {
+		ctx.Warningf("could not remove model allocation: %v", err)
+	}
+
+	budgetClient := getBudgetAPIClient(bakeryClient)
+
+	resp, err := budgetClient.DeleteAllocation(modelDetails.ModelUUID)
```

Comment:
Making this call on CLI means it's not part of the transaction. We could be in situation where model destruction succeeded but allocation was not deleted, erred out for some reason.
Is it acceptable?